src/components/CheckAuth/CheckAuth.js

```diff
@@ -0,0 +1,301 @@
+  margin-top: auto;
+  height: 100%;
+  flex: 1;
+`;
+
+const ACTIVE_APP_STATE = 'active';
+const BACKGROUND_APP_STATE = 'background';
+
+class CheckAuth extends React.Component<Props, State> {
+  static defaultProps = {
+    revealMnemonic: false,
+  };
+  state = {
+    biometricsShown: false,
+    lastAppState: AppState.currentState,
+    showPin: false,
+  };
+
+  componentDidMount() {
+    addAppStateChangeListener(this.handleAppStateChange);
+    const {
+      useBiometrics, revealMnemonic, enforcePin, modalProps,
+    } = this.props;
+    const { lastAppState } = this.state;
+
+    // do nothing if auth isn't supposed to be checked
+    if (modalProps && !get(modalProps, 'isVisible')) return;
+
+    if (useBiometrics
+      && !revealMnemonic
+      && lastAppState !== BACKGROUND_APP_STATE
+    ) {
+      this.showBiometricLogin();
+    } else if (lastAppState !== BACKGROUND_APP_STATE && !enforcePin) {
+      this.checkPrivateKey();
+    }
+  }
+
+  // special case for modals
+  componentDidUpdate = (prevProps: Props) => {
+    const { modalProps } = this.props;
+    if (!modalProps || !prevProps.modalProps) return;
+    if (modalProps.isVisible && !prevProps.modalProps.isVisible) {
+      this.checkPrivateKey();
+    }
+  };
+
+  hideModal(modalProps) {
+    if (!modalProps || !modalProps.isVisible) return;
+
+    if (modalProps.onModalHide) {
+      modalProps.onModalHide();
+    }
+
+    if (modalProps.onModalHidden) {
+      modalProps.onModalHidden();
+    }
+  }
+
+  checkPrivateKey = async (errorHandler?: Function) => {
+    const {
+      onPinValid, checkPrivateKey, modalProps, revealMnemonic, enforcePin,
+    } = this.props;
+    if (enforcePin) {
+      this.setState({ showPin: true });
+    } else {
+      const keychainData = await getKeychainDataObject(errorHandler);
+      if (keychainData) {
+        const { privateKey, mnemonic } = keychainData;
+        removeAppStateChangeListener(this.handleAppStateChange);
+        checkPrivateKey(privateKey, (_, wallet) => onPinValid(_,
+          revealMnemonic ? { ...wallet, mnemonic } : wallet));
+        this.hideModal(modalProps);
+      } else {
+        if (errorHandler) errorHandler();
+        this.setState({ showPin: true });
+      }
+    }
+  };
+
+  handleAppStateChange = (nextAppState: string) => {
+    const { useBiometrics, revealMnemonic } = this.props;
+    const { lastAppState } = this.state;
+    if (nextAppState === ACTIVE_APP_STATE
+      && lastAppState === BACKGROUND_APP_STATE
+      && useBiometrics
+      && !revealMnemonic) {
+      this.showBiometricLogin();
+    }
+    this.setState({ lastAppState: nextAppState });
+  };
+
+  showBiometricLogin() {
+    const { biometricsShown } = this.state;
+    if (biometricsShown) return;
+    this.setState({ biometricsShown: true }, () => {
+      this.checkPrivateKey(() => this.setState({ biometricsShown: false }));
+      this.setState({ biometricsShown: false });
+    });
+  }
```

Comment:
Just to follow the pattern throughout components.
```suggestion
  showBiometricLogin = () => {
    const { biometricsShown } = this.state;
    if (biometricsShown) return;
    this.setState({ biometricsShown: true }, () => {
      this.checkPrivateKey(() => this.setState({ biometricsShown: false }));
      this.setState({ biometricsShown: false });
    });
  };
```